src/Microsoft.IdentityModel.Protocols.WsFederation/Configuration/WsFederationConfigurationValidator.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.IdentityModel.Xml;
+using static Microsoft.IdentityModel.Logging.LogHelper;
+
+namespace Microsoft.IdentityModel.Protocols.WsFederation
+{
+    /// <summary>
+    /// Defines a class for validating the WsFederationConfiguration.
+    /// </summary>
+    public class WsFederationConfigurationValidator : IConfigurationValidator<WsFederationConfiguration>
+    {
+        /// <summary>
+        /// Validates a WsFederationConfiguration.
+        /// </summary>
+        /// <param name="configuration">WsFederationConfiguration to validate</param>
+        /// <returns>A <see cref="ConfigurationValidationResult"/> that contains validation result.</returns>
+        /// <exception cref="ArgumentNullException">If the provided configuration is null</exception>
+        public ConfigurationValidationResult Validate(WsFederationConfiguration configuration)
+        {
+            if (configuration == null)
+                throw LogArgumentNullException(nameof(configuration));
+
+            if (string.IsNullOrWhiteSpace(configuration.Issuer))
+            {
+                return new ConfigurationValidationResult
+                {
+                    ErrorMessage = LogMessages.IDX22700,
+                    Succeeded = false
+                };
+            }
```

Comment:
Consider doing all these checks up front and use StringBuilder to build out an error message. In the current model a developer may face one issue, update, face another issue, update...etc. Getting all the errors up front allows for more actionable errors/fewer iterations of debugging. Also less code maintenance costs IMO since if we decide to add another check it doesn't require another log message.

Reply:
The current code assume some of the checks before it. While is not a bad idea it requires a refactor of the whole method. If you dont mind I would like to postpone this improvement.